src/model/analysis.ts

```diff
@@ -82,6 +88,22 @@ export class Analysis implements Queryable {
   }
 
   /**
+   * Simplified version of getDocument which simply throws an error when asked
+   * for a missing document.
+   */
+  getDocumentOrDie(url: string): Document {
```

Comment:
I think this should be viewed as temporary until we have getDocument return a better result class that makes it easier to handle warnings.